Sources/SpelledPitch/PitchSpeller/Wetherfield/UndirectedGraphSchemeProtocol.swift

```diff
@@ -8,3 +8,24 @@
 import DataStructures
 
 public protocol UndirectedGraphSchemeProtocol: GraphSchemeProtocol where Edge == UnorderedPair<Node> { }
+
+extension UndirectedGraphSchemeProtocol where Self: WeightedGraphSchemeProtocol {
+    
+    func directed <D> () -> D where
+        D: DirectedGraphSchemeProtocol & WeightedGraphSchemeProtocol,
+        D.Weight == Weight,
+        D.Node == Node
+    {
+        return .init { edge in self.weight(from: edge.a, to: edge.b) }
+    }
+}
+
+extension UndirectedGraphSchemeProtocol where Self: UnweightedGraphSchemeProtocol {
+    
+    func directed <D> () -> D where
+        D: DirectedGraphSchemeProtocol & UnweightedGraphSchemeProtocol,
```

Comment:
Likewise, are there any types other than `DirectedScheme` which conform to both `DirectedGraphSchemeProtocol & UnweightedGraphSchemeProtocol`?

Reply:
ditto